### ISSUE TITLE
Take ownership when restarting

### ISSUE THREAD

**Issue by thomastaylor312** [deislabs/krustlet#127]
Right now when krustlet restarts, it gets 2 errors: 
```
Error creating node: ApiError: nodes "krustlet" already exists
Failed to create lease: ApiError: leases.coordination.k8s.io "krustlet" already exists
```

This causes it to stop updating its lease and shows the node as not ready. If the lease already exists, we need to continue onward and just patch the lease instead

**Comment by bacongobbler** [comment]
Could we run into an issue if someone starts two instances of krustlet-wasi with this change?

**Comment by bacongobbler** [comment]
That seems like a good middleground. Sounds good to me!

**Comment by thomastaylor312** [comment]
After looking at the code, node adoption is already in there, so I am going to create a separate issue for 0.2.0 that has us doing some more sane checks for adopting a node

**Comment by thomastaylor312** [comment]
Closed by #145 and superseded with #150
